.github/workflows/node.js.yml

```diff
@@ -10,20 +10,80 @@ on:
     branches: [ master ]
 
 jobs:
+# building job
   build:
-
     runs-on: ubuntu-latest
+    name: Build
 
     strategy:
       matrix:
         node-version: [10.x, 12.x, 14.x, 15.x]
 
     steps:
     - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
+    - name: Building on Node.js ${{ matrix.node-version }} 🛠️
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
+    - run: npm ci # equivalent of npm ci
+    - run: npm build
+      env:
+        CI: true
+# testing job
+  test:
+    runs-on: ubuntu-latest
+    name: Test
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Testing on node ${{ matrix.node-version }} 🧪
+      uses: actions/setup-node@v1
+      with:
+          node-version: ${{ matrix.node-version }}
     - run: npm ci
-    - run: npm run build --if-present
     - run: npm test
+# # coverage uploading job
+#   coverage:
+#     runs-on: ubuntu-latest
+#     name: Coverage
+#     needs: test
+
+#     steps:
+#     - name: Checkout
+#       uses: actions/checkout@v2
+#     - run: yarn coverage
+#     - name: Upload coverage to Codecov
+#       uses: codecov/codecov-action@v1
+#       with:
+#         token: ${{ secrets.CODECOV_TOKEN }}
+#         directory: ./tests/unit/coverage
+
+# release to npm job
+  release:
+    runs-on: ubuntu-latest
+    name: Release
+    needs: Test
+    # release only when pushed to master
+    if: github.event_name == 'push'
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+        # latest nodejs LTS
+          node-version: 14.x
+      - run: npm ci
+      - name: Preparing a release for npm and github 🚀
+      # more about this action here: https://github.com/cycjimmy/semantic-release-action
+        uses: cycjimmy/semantic-release-action@v2
+        with:
+          dry_run: true
```

Comment:
@sgtsquiggs @z3bi once you confirm that everything is working as expected you can remove the `dry_run` line for `semantic-release` to work properly